gammapy/spectrum/spectrum_analysis.py

```diff
@@ -184,7 +186,17 @@ def make_on_vector(self):
     def make_off_vector(self):
         """Make OFF `~gammapy.data.CountsSpectrum`
         """
-        off_list = self.event_list.select_sky_ring(self.target, self.irad, self.orad)
+        if self.off_type == "ring":
+            off_list = self.event_list.select_sky_ring(
+                self.target, self.irad, self.orad)
+        elif self.off_type == "reflected":
+            regions = find_reflected_regions_for_event_list(
+                self.event_list, self.exclusion, self.radius)
+            off_list = self.event_list.select_circular_regions(regions)
+            self.alpha = 1./len(regions)
+        else:
+            raise ValueError("Undefined background method: {}".format(self.off_type))
+            
```

Comment:
> Should this go in separate functions?

As you like.
In principle more modular is better, yes.
But if you don't have a second place where you'd like to use this functionality at the moment, it's also OK to keep here for now.